frontend/src/core/codemirror/go-to-definition/__tests__/commands.test.ts

```diff
@@ -31,30 +35,170 @@ afterEach(() => {
 
 describe("goToVariableDefinition", () => {
   test("selects the variable when it exists", async () => {
-    view = createEditor("#comment\nmyVar = 10\nprint(myVar)");
+    view = createEditor(`\
+#comment
+myVar = 10
+print(myVar)`);
     const result = goToVariableDefinition(view, "myVar");
 
     expect(result).toBe(true);
-    await new Promise((resolve) => requestAnimationFrame(resolve));
-    expect(view.state.selection.main.from).toBe(9);
-    expect(view.state.selection.main.to).toBe(9);
+    await tick();
+    expect(renderViewState(view)).toMatchInlineSnapshot(`
+      "
+      #comment
+      myVar = 10
+      ^
+      print(myVar)
+      "
+    `);
+  });
+
+  test("selects a function declaration", async () => {
+    view = createEditor(`\
+#comment
+def my_func():
+    pass
+
+print(my_func)`);
+    const result = goToVariableDefinition(view, "my_func");
+
+    expect(result).toBe(true);
+    await tick();
+    expect(renderViewState(view)).toMatchInlineSnapshot(`
+      "
+      #comment
+      def my_func():
+          ^
+          pass
+
+      print(my_func)
+      "
+    `);
+  });
+
+  test("selects outer-scope variable definition", async () => {
+    view = createEditor(`\
+x = 10
+
+def my_func(x):
+    print(x)
+
+print(x)`);
+    const result = goToVariableDefinition(view, "x");
+
+    expect(result).toBe(true);
+    await tick();
+    expect(renderViewState(view)).toMatchInlineSnapshot(`
+      "
+      x = 10
+      ^
+
+      def my_func(x):
+          print(x)
+
+      print(x)
+      "
+    `);
   });
 
-  test("does not select the variable when it does not exist", () => {
-    view = createEditor("#comment\nmyVar = 10;\nprint(myVar)");
+  test("selects outer-scope function declaration", async () => {
+    view = createEditor(`\
+def x():
+    print("hi")
+
+def my_func(x):
+    print(x)
+
+print(x)`);
+    const result = goToVariableDefinition(view, "x");
+
+    expect(result).toBe(true);
+    await tick();
+    expect(renderViewState(view)).toMatchInlineSnapshot(`
+      "
+      def x():
+          ^
+          print("hi")
+
+      def my_func(x):
+          print(x)
+
+      print(x)
+      "
```

Comment:
I switched these tests to use snapshotting because I find them a.) much easier to write and b.) the diffs are so much nicer when debugging:

<img width500 src="https://github.com/user-attachments/assets/7196ee06-2150-4cb4-9731-a4e8775ebdaf" />